xbmc/main/win32/MessagePrinter.cpp

```diff
@@ -0,0 +1,51 @@
+/*
+*      Copyright (C) 2005-2015 Team XBMC
+*      http://xbmc.org
+*
+*  This Program is free software; you can redistribute it and/or modify
+*  it under the terms of the GNU General Public License as published by
+*  the Free Software Foundation; either version 2, or (at your option)
+*  any later version.
+*
+*  This Program is distributed in the hope that it will be useful,
+*  but WITHOUT ANY WARRANTY; without even the implied warranty of
+*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+*  GNU General Public License for more details.
+*
+*  You should have received a copy of the GNU General Public License
+*  along with XBMC; see the file COPYING.  If not, see
+*  <http://www.gnu.org/licenses/>.
+*
+*/
+
+#include "MessagePrinter.h"
+#include "CompileInfo.h"
+
+#include <windows.h>
+
+void CMessagePrinter::DisplayMessage(const std::string& message)
+{
+  MessageBox(NULL, message.c_str(), CCompileInfo::GetAppName(), MB_OK | MB_ICONINFORMATION);
+}
+
+void CMessagePrinter::DisplayWarning(const std::string& warning)
+{
+  MessageBox(NULL, warning.c_str(), CCompileInfo::GetAppName(), MB_OK | MB_ICONWARNING);
+}
+
+void CMessagePrinter::DisplayError(const std::string& error)
+{
+  MessageBox(NULL, error.c_str(), CCompileInfo::GetAppName(), MB_OK | MB_ICONERROR);
+}
+
+void CMessagePrinter::DisplayHelpMessage(const std::vector<std::pair<std::string, std::string>>& help)
+{
+  //very crude implementation, pretty it up when possible
+  std::string message;
+  for (auto& line : help)
```

Comment:
This will actually be const because the vector is const it will resolve to the const iterators. With that said i missed to put a const there :)